xtras/src/supervisor.rs

```diff
@@ -28,18 +28,58 @@ struct Metrics {
     pub num_panics: u64,
 }
 
-impl<T, R> Actor<T, R>
+struct UnitReason {}
+
+impl fmt::Display for UnitReason {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "()")
+    }
+}
+
+impl From<()> for UnitReason {
+    fn from(_: ()) -> Self {
+        UnitReason {}
+    }
+}
+
+impl<T> Actor<T, UnitReason>
+where
+    T: xtra::Actor<Stop = ()>,
+{
+    /// Construct a new supervisor for an [`Actor`] with an [`xtra::Actor::Stop`] value of `()`.
+    ///
+    /// The actor will always be restarted if it stops. If you don't want this behaviour, don't use
+    /// a supervisor. If you want more fine-granular control in which circumstances the actor
+    /// should be restarted, set [`xtra::Actor::Stop`] to a more descriptive value and use
+    /// [`Actor::with_policy`].
+    pub fn new(ctor: impl (Fn() -> T) + Send + 'static) -> (Self, Address<T>) {
```

Comment:
👍 I like the design with `new` and `with_policy`. Just in case you didn't now, `new` is unused outside of this crate. But I think that's okay.

Reply:
Yep, we still need to adopt it!